dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+                }
+            }
+
+            /**
+             * Removes the filter by showing all options and optgroups in the select.
+             * @param {jQuery} $selects - the selects to remove the filter of
+             */
+            function removeFilter($selects) {
+                extendedShow($selects.find('option, optgroup'));
+            }
+
+            /**
+             * Debug function to measure performance for functions.
+             * Takes a start timestamp and calculates duration using the end which is now.
+             * @param {number} startTimestamp - the start of the measure activity
+             * @param {string} feature - the feature to be tested to better discern in the console log
+             */
+            function measurePerformance(startTimestamp, feature) {
+                var endTimestamp = performance.now();
+                var timeSpent = endTimestamp - startTimestamp;
+                console.log("Measuring " + feature + ": " + timeSpent);
+            }
+
+            /**
+             * Given a select and the option to return only selected options or all of them,
+             * this returns all requested options (but no invisible/hidden options)
+             * @param {jQuery} $select - the select to find options in
+             * @param {boolean} [onlySelected=true] - if true, returns only selected options; if false, returns all options
+             * @returns {jQuery} - the requested options
+             */
+            function getOptionsToMove($select, onlySelected) {
+                if (onlySelected === undefined) {
+                    onlySelected = true;
+                }
+                // FIXME: Unsure if this is good, as I could define and use my own hidden class
+                /** @type {jQuery} */
+                var $allVisibleOptions = $select.find("option:not(" + SELECTOR_HIDDEN + ")");
+                if (onlySelected) {
+                    return $allVisibleOptions.filter(":selected");
+                } else {
+                    return $allVisibleOptions;
+                }
+            }
+
+            /**
+             * Checks the two sides of the Multiselect and removes any duplicate options
+             * (and maybe empty optgroups) from the left side
+             * @param {jQuery} $left - the left select
+             * @param {jQuery} $right - the right selects
+             */
+            function removeDuplicateOptions($left, $right) {
+                // FIXME: What if one right element has the same option as another right element?
+                $right.find('option').each(function(index, rightOption) {
+                    /** @type {jQuery} */
+                    var $rightOption = $(rightOption);
+                    /** @type {jQuery} */
+                    var $parentOptgroup = $rightOption.parent("optgroup");
+                    /** @type {string} */
+                    var duplicateOptionSelector = 'option[value="' + ri
```

Comment:
This is not that ideal, as I don't give the instance as parameter, I have to get the instance using the source or destiniation select element.
The best way would be to have real private methods, but as JS doesn't have these and I didn't want to expose the module's own move function (because the user should use the providable callback, shouldn't he?)